app/models/manageiq/providers/vmware/infra_manager/inventory/collector.rb

```diff
@@ -344,6 +345,12 @@ def collect_cis_taggings(api_client)
     end
   end
 
+  # These are only collected for full refreshes, after a full they can be cleared
+  # to free up memory and prevent targeted refreshes from trying to map labels
+  def clear_cis_taggings
+    self.categories_by_id = self.tags_by_id = self.tag_ids_by_attached_object = nil
```

Comment:
These are collected [here](https://github.com/ManageIQ/manageiq-providers-vmware/pull/695/files#diff-1354967fd69afcb0caf8af2d735d4dc26246f72e37a195a456cc0608fbd184d0R336-R339) and used by the parser [here](https://github.com/ManageIQ/manageiq-providers-vmware/blob/1c1e503ca90fe4c7e20c23c466aef3d105e29755/app/models/manageiq/providers/vmware/infra_manager/inventory/parser/virtual_machine.rb#L370-L375)

Since we reuse the same collector object for targeted refreshes we were erroniously trying to map tags during a targeted refresh if a vm with a CIS tag got an update causing a failure due to the persister.tag_mapper not being initialized.